src/common/sr_utils.c

```diff
@@ -2547,6 +2552,42 @@ sr_create_uri_for_module(const struct lys_module *module, char **uri)
 }
 
 int
+sr_time_to_str(time_t time, char *buff, size_t buff_size)
+{
+    CHECK_NULL_ARG(buff);
+
+    strftime(buff, buff_size - 1, "%Y-%m-%dT%H:%M:%S%z", localtime(&time));
+    /* time buff ends in '+hhmm' but should be '+hh:mm' */
+    memmove(buff + strlen(buff) - 1, buff + strlen(buff) - 2, 3);
+    buff[strlen(buff) - 3] = ':';
+
+    return SR_ERR_OK;
+}
+
+int
+sr_str_to_time(char *time_str, time_t *time)
+{
+    struct tm tm = { 0, };
+    char *time_str_copy = NULL, *colon = NULL;
+
+    CHECK_NULL_ARG2(time_str, time);
+
+    time_str_copy = strdup(time_str);
+    CHECK_NULL_NOMEM_RETURN(time_str_copy);
+
+    /* time str ends in '+hh:mm' but should be '+hhmm' */
+    colon = strrchr(time_str_copy, ':');
+    memmove(colon, colon + 1, 2);
```

Comment:
Does the user have control over the string data? If yes, then this is an undefined behavior when the string doesn't contain colon.